src/validator/IndexScanValidator.cpp

```diff
@@ -0,0 +1,147 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "validator/IndexScanValidator.h"
+#include "planner/Query.h"
+#include "util/ExpressionUtils.h"
+#include "util/SchemaUtil.h"
+
+namespace nebula {
+namespace graph {
+
+Status IndexScanValidator::validateImpl() {
+    NG_RETURN_IF_ERROR(prepareFrom());
+    NG_RETURN_IF_ERROR(prepareYield());
+    NG_RETURN_IF_ERROR(prepareFilter());
+    return Status::OK();
+}
+
+Status IndexScanValidator::toPlan() {
```

Comment:
Maybe `LookUpValidator` is better.

Reply:
> Maybe `LookUpValidator` is better.

Emm ... It's uniformly named `IndexScan` , because it will also be used for other statements, not just lookup

Reply:
Validator is used for one statement.